persistence/jdbc/src/test/java/org/infinispan/persistence/jdbc/stringbased/JdbcStringBasedStoreTxFunctionalTest.java

```diff
@@ -0,0 +1,132 @@
+   }
+
+   @Override
+   protected EmbeddedCacheManager createCacheManager(boolean start, GlobalConfigurationBuilder global,
+         ConfigurationBuilder cb) {
+      // Make sure defaults are transactional as well for created configs
+      if (transactionalConfig) {
+         cb.transaction().transactionMode(TransactionMode.TRANSACTIONAL);
+      }
+      if (sharedConfig) {
+         ConfigurationBuilderHolder holder = new ConfigurationBuilderHolder(Thread.currentThread().getContextClassLoader(), global);
+         if (cb != null) {
+            String defaultName = JdbcStringBasedStoreTxFunctionalTest.class.getName() + "-default";
+            global.defaultCacheName(defaultName);
+            holder.newConfigurationBuilder(defaultName).read(cb.build());
+         }
+         global.transport().defaultTransport();
+         return TestCacheManagerFactory.createClusteredCacheManager(start, holder);
+      } else {
+         return super.createCacheManager(start, global, cb);
+      }
+   }
+
+   @Factory
+   public Object[] factory() {
+      return new Object[]{
+            new JdbcStringBasedStoreTxFunctionalTest().transactionalConfig(true).sharedConfig(false),
+            new JdbcStringBasedStoreTxFunctionalTest().transactionalConfig(false).sharedConfig(false),
+            new JdbcStringBasedStoreTxFunctionalTest().transactionalConfig(true).sharedConfig(true),
+            new JdbcStringBasedStoreTxFunctionalTest().transactionalConfig(false).sharedConfig(true),
+      };
+   }
+
+   @Override
+   protected String parameters() {
+      return " [transactionalConfig=" + transactionalConfig + ", sharedConfig=" + sharedConfig + "]";
+   }
+
+   @Override
+   public void testRemoveCacheWithPassivation() {
+      // Shared and purgeOnStartup don't mix
+      // Transactional and passivation don't mix
+      if (!sharedConfig && !transactionalConfig) {
+         super.testRemoveCacheWithPassivation();
+      }
+   }
+
+   @Override
+   public void testStoreByteArrays(Method m) throws PersistenceException {
+      // Shared and purgeOnStartup don't mix
+      // Transactional and passivation don't mix
+      if (!sharedConfig && !transactionalConfig) {
+         super.testStoreByteArrays(m);
+      }
+   }
+
+   public void testRollback() throws SystemException, NotSupportedException {
+      String cacheName = "testRollback";
+      ConfigurationBuilder cb = getDefaultCacheConfiguration();
+      createCacheStoreConfig(cb.persistence(), false);
+      cacheManager.defineConfiguration(cacheName, cb.build());
+
+      Cache<String, Object> cache = cacheManager.getCache(cacheName);
+
+      String key = "rollback-test";
+      assertNull(cache.get(key));
+
+      TransactionManager manager = cache.getAdvancedCache().getTransactionManager();
+
+      String value = "the-value";
+      manager.begin();
+      cache.put(key, value);
+      assertEquals(value, cache.get(key));
+      manager.rollback();
+
+      assertNull(cache.get(key));
```

Comment:
Shouldn't we also check that "rollback-test" key is not present in the store after rollback?

Reply:
I personally don't think it is worth it. The `cache.get` should be doing that already. None of the rest of the tests in this class hierarchy go directly at the store either.